src/clue.js

```diff
@@ -2,29 +2,125 @@
+    color: 'white'
+  },
+  {
+    firstName: 'Victor', //profPlum
+    lastName: 'Plum',
+    occupation: 'Designer',
+    age: 22,
+    description: 'Billionaire video game designer',
+    image:
+      'https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg',
+    color: 'purple'
+  },
+  {
+    firstName: 'Kasandra', //missScarlet
+    lastName: 'Scarlet',
+    occupation: 'Actor',
+    age: 31,
+    description: 'She is an A-list movie star with a dark past',
+    image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+    color: 'red'
+  },
+  {
+    firstName: 'Eleanor', // mrsPeacock
+    lastName: 'Peacock',
+    occupation: 'Socialité',
+    age: 36,
+    description:
+      'She is from a wealthy family and uses her status and money to earn popularity',
+    image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+    color: 'blue'
+  },
+  {
+    firstName: 'Jack', // mrMustard
+    lastName: 'Mustard',
+    occupation: 'Retired Football player',
+    age: 62,
+    description:
+      'He is a former football player who tries to get by on his former glory',
+    image:
+      'https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg',
+    color: 'yellow'
+  }
+];
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = [
+  { name: 'Dining Room' },
+  { name: 'Conservatory' },
+  { name: 'Kitchen' },
+  { name: 'Study' },
+  { name: 'Library' },
+  { name: 'Billiard Room' },
+  { name: 'Lounge' },
+  { name: 'Ballroom' },
+  { name: 'Hall' },
+  { name: 'Spa' },
+  { name: 'Living Room' },
+  { name: 'Observatory' },
+  { name: 'Theater' },
+  { name: 'Guest House' },
+  { name: 'Patio' }
+];
 
 // Weapons Array
 
-const weaponsArray = [];
-
+const weaponsArray = [
+  { name: 'rope', weight: 10 },
+  { name: 'knife', weight: 8 },
+  { name: 'candlestick', weight: 2 },
+  { name: 'dumbbell', weight: 30 },
+  { name: 'poison', weight: 2 },
+  { name: 'axe', weight: 15 },
+  { name: 'bat', weight: 13 },
+  { name: 'trophy', weight: 25 },
+  { name: 'pistol', weight: 20 }
+];
 
 // ITERATION 2
 
-function selectRandom() {}
-
-function pickMystery() {}
+function selectRandom(arr) {
+  let result = arr[Math.floor(Math.random() * arr.length)];
+  return result;
+}
 
+function pickMystery() {
+  let suspect = selectRandom(suspectsArray);
+  let weapon = selectRandom(weaponsArray);
+  let room = selectRandom(roomsArray);
+  return { suspect, weapon, room };
+}
+// I can't believe I got this one right!!
+// I was overcomplicating at first.
 
 // ITERATION 3
 
-function revealMystery() {}
-
-
+function revealMystery({}) {
+  //I really don't know what to write here (it's probably very simple...)
+  //I also don't know what's missing in the line below
+  return `${suspectsArray.firstName} ${suspectsArray.lastName} killed Mr. Body using the ${weaponsArray.name} in the ${roomsArray.name}!`;
+}
```

Comment:
Almost!

- your function received an object -> so you want to pass any name `function revealMystery(myObject)`
- then before you start any logic, I advise console.log information that you receive `console.log(myObject)`, so you know the structure of your data
- Now I believe you should be able to fix line122 accordingly. Let me know if still not clear! Karina